elide-graphql/src/main/java/com/yahoo/elide/graphql/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,410 @@
+        String entityName = entityField.getName();
+        Class<?> entityType = entityDictionary.getEntityClass(entityName);
+        if (entityType == null) {
+            throw new InvalidEntityBodyException(String.format("Unknown entity {%s}.", entityName));
+        }
+
+        final EntityProjection entityProjection = EntityProjection.builder()
+                .dictionary(entityDictionary)
+                .type(entityType)
+                .build();
+
+        entityField.getArguments().forEach(argument -> addArgument(argument, entityProjection));
+
+        SelectionSet edges = entityField.getSelectionSet();
+        SelectionSet node = addEdges(edges, entityProjection);
+        SelectionSet fields = addNode(node, entityProjection);
+        fields.getSelections().forEach(fieldSelection -> addField(fieldSelection, entityProjection));
+
+        return entityProjection;
+    }
+
+    /**
+     * Get information of an {@link EntityProjection} from Elide 'edges' container and get the Elide 'node' container
+     *
+     * @param edges Elide 'edges' container
+     * @param entityProjection projection that links with this container
+     * @return Elide 'node' container
+     */
+    private static SelectionSet addEdges(SelectionSet edges, EntityProjection entityProjection) {
+        if (edges.getSelections().size() != 1) {
+            throw new InvalidEntityBodyException("Entity selection must have one 'edges' graphQL field node.");
+        }
+
+        Selection edgesSelection = edges.getSelections().get(0);
+        if (!(edgesSelection instanceof Field)) {
+            throw new InvalidEntityBodyException("Edges selection must be a graphQL field.");
+        }
+
+        return ((Field) edgesSelection).getSelectionSet();
+    }
+
+    /**
+     * Get information of an {@link EntityProjection} from Elide 'node' container and get fields from the container
+     *
+     * @param node Elide 'node' container
+     * @param entityProjection projection that links with this container
+     * @return Fields in the 'node' container
+     */
+    private static SelectionSet addNode(SelectionSet node, EntityProjection entityProjection) {
+        if (node.getSelections().size() != 1) {
+            throw new InvalidEntityBodyException("Edges selection must have one 'node' graphQL field node.");
+        }
+
+        Selection nodeSelection = node.getSelections().get(0);
+        if (!(nodeSelection instanceof Field)) {
+            throw new InvalidEntityBodyException("Node selection must be a graphQL field.");
+        }
+
+        return ((Field) nodeSelection).getSelectionSet();
+    }
+
+    /**
+     * Add a {@link Selection} that represents a field/fragment to an {@link EntityProjection}
+     *
+     * @param fieldSelection field/fragment to add
+     * @param parentProjection projection that has this field/fragment
+     */
+    private void addField(Selection fieldSelection, final EntityProjection parentProjection) {
```

Comment:
addField method should only handle Field arguments.  The parent caller can determine the type and call the appropriate method.  That should keep the lines per function smaller.